### ISSUE TITLE
Yoda flags Hash subset syntax

### ISSUE THREAD

**Issue by mperham** [rubocop/rubocop#10838]
Ruby allows you to use `<=` to check for a subset of a Hash which I'm using as a test assertion (as the hash also includes some racy, unpredictable data):

```ruby
hash = {time: "2022-07-22T22:03:00Z", p: 1, f: 0, some: "unpredictable timing"} 
assert({time: "2022-07-22T22:03:00Z", p: 1, f: 0} <= hash)
```

https://stackoverflow.com/a/35135438/1494519

This syntax is flagged by Style/YodaCondition. Is this intended?

Original issue: https://github.com/testdouble/standard/issues/442

**Comment by dvandersluis** [comment]
This is interesting, I didn't know about this syntax for hashes.

I'm not 100% sold that it shouldn't be flagged by the cop though. The idea is that a yoda condition is where the variable is on the right and the literal on the left, which is the case here. It'd be more obvious with equality perhaps:

```ruby
{time: "2022-07-22T22:03:00Z", p: 1, f: 0} == hash
```

But since this can be rewritten without the yoda condition with the same behaviour:

```ruby
hash >= {time: "2022-07-22T22:03:00Z", p: 1, f: 0} 
```

... I'm not sure if this is truly a false positive. @rubocop/rubocop-core does anyone thing we should add a special case here for hashes with `>`, `>=`, `<` and `<=`?

**Comment by dvandersluis** [comment]
FWIW this is inline with behaviour for other non-equality comparison operators and `Style/YodaCondition`.

```ruby
# bad
5 < foo

# good
foo >= 5
```

etc.

**Comment by mperham** [comment]
My code copies the standard `assert expected, actual` ordering.

**Comment by koic** [comment]
I think this is not an issue of `Style/YodaCondition` cop and it is an issue of testing framework usage. For example, in Minitest can be written:

```ruby
hash = {time: "2022-07-22T22:03:00Z", p: 1, f: 0, some: "unpredictable timing"} 
assert_operator({time: "2022-07-22T22:03:00Z", p: 1, f: 0}, :<=, hash)
```

I'm going close this issue because it's maybe a testing code specific issue.
